modules/core/src/null-loader.ts

```diff
@@ -18,6 +18,7 @@ export const NullLoader: LoaderWithParser = {
   parseSync: (arrayBuffer) => arrayBuffer,
   // @ts-ignore
   parseInBatches: async (asyncIterator) =>
+    // @ts-ignore
     (async function* parseInBatches() {
       // @ts-ignore
       yield* asyncIterator;
```

Comment:
I admit that the lack of compatibility between these two iterator types is confusing, however I am not convinced that just "hiding" the problem this way is better. Our return type is basically promising one thing, and we actually return another, which seems rather evil.

Perhaps

```js
    (async function* parseInBatches() {
      for await (const batch of asyncIterator) {
      yield batch;
   }
}
```

Reply:
> I am not convinced that just "hiding" the problem this way is better.
> 
Well I didn't really dive into it because it was already surrounded by two `// @ts-ignores`, but now that you presented the challenge I tried to solve the types. Updated with the PR with what seems to work.